src/main/java/fi/aalto/cs/intellij/actions/RequiredPluginsCheckerAction.java

```diff
@@ -0,0 +1,211 @@
+    if (missingOrDisabledPluginNames.size() > 0) {
+      getAvailablePluginsFromMainRepo();
+      missingOrDisabledIdeaPluginDescriptors = new ArrayList<>();
+      missingOrDisabledPluginNames.forEach((name, id) -> {
+        PluginId missingPluginId = PluginId.getId(id);
+        availableIdeaPluginDescriptors
+            .stream()
+            .filter(
+                availableDescriptor -> availableDescriptor.getPluginId().equals(missingPluginId)
+            )
+            .findFirst()
+            .ifPresent(missingOrDisabledIdeaPluginDescriptors::add);
+      });
+    }
+  }
+
+  /**
+   * Get the list of all the available in the main JetBrains plugin repository.
+   * <p>
+   * Note! The descriptors for the not installed plugins do not exist in IJ until now.
+   */
+  private void getAvailablePluginsFromMainRepo() {
+    try {
+      availableIdeaPluginDescriptors = RepositoryHelper.loadPlugins(new BgProgressIndicator());
+    } catch (IOException ex) {
+      logger.error("Could not retrieve plugins data from main repository.", ex);
+    }
+  }
+
+  /**
+   * Sorts required plugins into missing and disabled and shows respective notifications.
+   */
+  private void checkPluginsStatusAndNotify() {
+    List<IdeaPluginDescriptor> missingPluginDescriptors = new ArrayList<>();
+    List<IdeaPluginDescriptor> disabledPluginDescriptors = new ArrayList<>();
+
+    for (IdeaPluginDescriptor descriptor : missingOrDisabledIdeaPluginDescriptors) {
+      if (!PluginManager.isPluginInstalled(descriptor.getPluginId())) {
+        missingPluginDescriptors.add(descriptor);
+      } else if (PluginManager.isDisabled(descriptor.getPluginId().getIdString())) {
+        disabledPluginDescriptors.add(descriptor);
+      }
+    }
+
+    if (missingPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsInstallation(missingPluginDescriptors);
+    } else if (disabledPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsEnabling(disabledPluginDescriptors);
+    }
+  }
+
+  /**
+   * Notify with with an option to enable all the required plugins.
+   */
+  private void notifyAndSuggestPluginsEnabling(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
+        "A+",
+        "A+",
+        "Some plugins must be and enabled for the A+ plugin to work properly "
+            + getPluginsNamesString(descriptors) + ".",
+        NotificationType.WARNING);
+
+    notification.addAction(new NotificationAction(
+        "Enable the required plugin(s) (" + getPluginsNamesString(descriptors) + ").") {
+
+      /**
+       * Activate all the required plugins.
+       */
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+        descriptors.forEach(descriptor -> Objects
+            .requireNonNull(PluginManager.getPlugin(descriptor.getPluginId())).setEnabled(true));
+        notification.expire();
+      }
+    });
```

Comment:
**potential testing**: how do you feel about the anonymous actions (as it might come as an issue if we ever decide to test actions)?

Reply:
If we want to unit test those actions they can be easily refactored into classes. Until there are unit tests, I see no reason to do that. IMO this is fine. :+1: